<launch>

	<node name="pointcloud2_to_depth" pkg="depth_tools" type="pointcloud2_to_depth_node" respawn="false" output="screen">
		<remap from="/cloud_in" to="/tilt_scan_cloud" />
		<remap from="/depth/image_raw" to="/laser/depth/image_raw" />

		<param name="world_frame" type="string" value="odom_combined" />
		<param name="camera_frame" type="string" value="laser_tilt_static_mount_link" />

		<param name="max_depth" type="double" value="10.0" /> <!-- Cannot be more than 65m (or ushort in mm.) -->

		<param name="camera_width" type="int" value="1920" />
		<param name="camera_height" type="int" value="1080" />

		<param name="postprocess_x" type="bool" value="true" />
		<param name="x_max_gap" type="int" value="25" />
		<param name="postprocess_y" type="bool" value="true" />
		<param name="y_max_gap" type="int" value="20" />

	</node>

	<node name="fake_laser_rgb" pkg="depth_tools" type="fake_laser_rgb_node" respawn="false" output="screen">
		<remap from="/image_raw" to="/laser/rgb/image_raw" />
		<param name="frame_rate" type="int" value="20" />
		<param name="camera_width" type="int" value="1920" />
		<param name="camera_height" type="int" value="1080" />
		<param name="camera_frame" type="string" value="laser_tilt_static_mount_link" />
	</node>


  <node pkg="nodelet" type="nodelet" name="cloudify" args="load depth_image_proc/point_cloud_xyz  record_player_manager --no-bond">
    <remap from="image_rect" to="laser/depth/image_raw"/>
    <remap from="camera_frame" to="laser/depth/camera_info"/>
    <remap from="points" to="/laser/depth/points"/>
    <!-- <remap from="rgb/camera_info" to="$(arg cloud_input_ns)/camera_info"/> -->
  </node>
</launch>
